### ISSUE TITLE
add option to AtmosphereProfile.line_of_sight_integral to include curvature of Earth

### ISSUE THREAD

**Issue by maxnoe** [cta-observatory/ctapipe#2294]
**Please describe the use case that requires this feature.**

This is important for high-zenith-angle observations

**Describe the solution you'd like**

Include a formula / numeric method for computing the line of sight integral assuming spherical earth.

**Describe alternatives you've considered**

There is not really an alternative, for high zenith angles, the current solution will produce nonsensical results.

**Additional context**

This problem will have a solution somewhere, didn't check yet

**Comment by kosack** [comment]
yes, I only implemented the simplest case. We should probably at least implement the Corsika model, which I think assumes a Earth is a perfect sphere with  6371 km radius, and the atmosphere is assumed to be spherically symmetric around the Earth's center.

**Comment by kosack** [comment]
Actually, no I'm wrong, looking in the corsika paper (Heck et al 1998), they say  (paraphrasing):  _only a flat atmosphere is used... When the zenith angle is ≥75°, the difference between flat and spherical is no longer negligible and an approximation is made._  They approximate nearly horizontal showers by setting the zenith angle to 0 and using a flat atmosphere profile that is basically the value at the observation level.   

This is also backed up by Konrad's presentation here http://hess.in2p3.fr/aaa/02__shower.pdf

**Comment by kosack** [comment]
Since CTA's requirements only are to observe to 60°, it may not be necessary to support spherical atmosphere integration.  However, for cases like MAGIC observing near or below the horizon (due to being on a mountain), it would be necessary, but perhaps that is too special a case to require implementing something fancier.

**Comment by kosack** [comment]
So perhaps we can close this
